### ISSUE TITLE
Revisit how the worker scopes are implemented

### ISSUE THREAD

**Issue by isudana** [ballerinalang/ballerina#2104]
In the current implementation since workers resides in the package scope, even two workers with same name placed inside two different functions cannot co-exists.

**Comment by lankavitharana** [comment]
can we just use the same way to register the workers as blockstmt and just change the resolving part so that workers can only see what it needs to see?

**Comment by chanakaudaya** [comment]
@isudana @lankavitharana Have you guys tried out a sample with what you have mentioned? I tried the following sample and it worked for me. In this sample, I have 2 workers within the same package (same file) with the same name and it worked without any issue.
```
package test;
import ballerina.lang.system;
import ballerina.lang.messages;

const int index = 12;

function main(string[] args) {
  worker sampleWorker (message m)  {
    json j;
    j = `{"name":"chanaka"}`;
    messages:setJsonPayload(m, j);
    system:println("constant value is " + index);
    printHello(m);
    reply m;
  }
  message result;
  message msg = {};
  msg -> sampleWorker;
  system:println("After worker");
  result <- sampleWorker;
  string s = messages:getStringPayload(result);
  printHello(result);
  printHello2(result);
  //workerCaller("sampleWorker");
  system:println(s);

}

function printHello(message msg) {
  worker sampleWorker (message m)  {
    json j;
    j = `{"name":"hello"}`;
    messages:setJsonPayload(m, j);
    system:println("constant value is " + index);
    //system:println("function variable " + aa);
    reply m;
  }
  message response = {};
  msg -> sampleWorker;
  system:println("Worker started at printHello");
  response <- sampleWorker;
  system:println(messages:getStringPayload(response));
}

function printHello2(message msg) {
  worker sampleWorker (message m)  {
    json j;
    j = `{"name":"hello"}`;
    messages:setJsonPayload(m, j);
    system:println("constant value is " + index);
    //system:println("function variable " + aa);
    reply m;
  }
  message response = {};
  msg -> sampleWorker;
  system:println("Worker started at printHello2");
  response <- sampleWorker;
  system:println(messages:getStringPayload(response));
}

```

**Comment by hasithaa** [comment]
This is correctly fixed in 0.94
